LF: Kill deprecated Transaction versions (#7785)
digital-asset/daml: daml-lf/spec/transaction.rst

This table lists every version of this specification in ascending order
(oldest first).

> the *dev* version is used as staging area for the next stable transaction
> version. The "dev" version can be changed freely without compatibility
> considerations to prior "dev" versions.

Reviewer: ```suggestion
The *"dev" version* is used as a staging area for the next stable transaction
version. The "dev" version can be changed freely without considering
compatibility with prior "dev" versions.
```

> Supported for transaction versions 1 to 9 has been dropped since 2020-11-02.
> This breaking changes does not impact ledger created with DAML SDK 1.0.0 or
> latter.

Reviewer: ```suggestion
Support for transaction versions 1 to 9 was dropped on 2020-11-02.
This breaking change does not impact ledgers created with SDK 1.0.0 or
later.
```
